plasmapy/classes/sources/coils.py

```diff
@@ -0,0 +1,166 @@
+
+        Parameters
+        ----------
+        r : u.m
+            Position; can be a (3,) array or a (N, 3) array.
+
+        Returns
+        -------
+        electric field array of the same shape as the input.
+        """
+        return u.Quantity(self._interpolate_E(r.si.value), E_unit)
+
+    def _interpolate_B(self, r: np.ndarray):
+        B = np.zeros(r.shape, dtype=float)
+        for ms in self.magnetostatics:
+            # TODO - I had trouble calling the function otherwise...
+            field = ms._magnetic_field(r, ms.pt, ms.dl, ms.current, ms.w)
+            B += field
+        return B
+
+    def interpolate_B(self, r: u.m):
+        """
+        Returns the magnetic field from all the MagnetoStatics, with units.
+
+        Parameters
+        ----------
+        r : u.m
+            Position; can be a (3,) array or a (N, 3) array.
+
+        Returns
+        -------
+        Magnetic field array of the same shape as the input.
+        """
+        return u.Quantity(self._interpolate_B(r.si.value), u.T, copy=False)
+
+    def __repr__(self):
+        names = []
+        for ms in self.magnetostatics:
+            name = f"{ms.__class__.__name__}({ms.current * u.A})"
+            names.append(name)
+        from collections import Counter
+
+        items = []
+        for key, value in Counter(names).items():
+            items.append(f"{value} x {key}")
+        return f"{self.__class__.__name__}({', '.join(items)})"
+
+    @classmethod
+    def is_datasource_for(cls, **kwargs):
+        match = "interpolate_B" in kwargs.keys()
+        return match
+
+    def visualize(self, figure=None):  # coverage: ignore
+        """
+        Visualizes the set of coils in 3D using PyVista.
+
+        Parameters
+        ----------
+        figure : pyvista.BasePlotter, optional
+            If not provided, a new plotter is created.
+
+        Returns
+        -------
+        the same `pyvista.BasePlotter` instance as (if) provided;
+        this allows method chaining.
+        """
+        import pyvista as pv
+
+        if figure is None:
+            fig = pv.Plotter()
+        else:
+            fig = figure
+
+        for ms in self.magnetostatics:
+            ms.visualize(fig)
+
+        return fig
+
+    @classmethod
+    def toykamak(
+        cls,
+        minor_radius: u.m = 0.3 * u.m,
+        radius: u.m = 1 * u.m,
+        main_current: u.A = 15 * u.MA,
+        coil_currents=None,
+    ):
+        """
+        Creates a set of coils for the Toykamak model.
+
+        Parameters
+        ----------
+        minor_radius : u.m
+            radius of the poloidal coils.
+        radius : u.m
+            major radius of the single coil representing the plasma
+        main_current : u.A
+            current through the single plasma-representing coil
+        coil_currents :  Iterable[u.A]
+            list of currents through the coils. By default, 8 15-MA coils are created.
```

Comment:
```suggestion
            list of currents through the coils. By default, eight 15-MA coils are created.
```
For readability...maybe `15 MA` instead of `15-MA` too, unless using the hyphen is standard format.

Reply:
I edited your suggestions to remove the hyphen. Laziness overload.